bftengine/src/bftengine/ReplicaForStateTransfer.hpp

```diff
@@ -46,6 +48,14 @@ class ReplicaForStateTransfer : public IReplicaForStateTransfer, public ReplicaB
  protected:
   virtual void onTransferringCompleteImp(uint64_t checkpointNumberOfNewState) = 0;
 
+  template <typename T>
+  void consensusMessageHandler(MessageBase* msg) {
```

Comment:
Please move to cpp, keep only decleration.
That's why we have cpp. in that case hpp is used for declerations, cpp for defnitions.